clar2wasm/src/wasm_generator.rs

```diff
@@ -2016,6 +2152,116 @@ impl ASTVisitor for WasmGenerator<'_> {
 
         Ok(builder)
     }
+
+    fn visit_unwrap_panic<'b>(
+        &mut self,
+        mut builder: InstrSeqBuilder<'b>,
+        _expr: &SymbolicExpression,
+        input: &SymbolicExpression,
+    ) -> Result<InstrSeqBuilder<'b>, InstrSeqBuilder<'b>> {
+        // There must be either an `optional` or a `response` on the top of the
+        // stack. Both use an i32 indicator, where 0 means `none` or `err`. In
+        // both cases, if this indicator is a 0, then we need to early exit.
+
+        // Get the type of the input expression
+        let input_ty = self
+            .get_expr_type(input)
+            .expect("try input expression must be typed")
+            .clone();
+
+        match &input_ty {
+            TypeSignature::OptionalType(val_ty) => {
+                // For the optional case, e.g. `(unwrap-panic (some 1))`, the stack
+                // will look like:
+                // 1 -- some value
+                // 1 -- indicator
```

Comment:
Oof, I did not think about that, I thought it would come in the other way around.

Reply:
Yeah, I wonder if it makes sense to switch the order, since we're dealing with a stack machine... I'd have to think about that some more. Let's save it for a different PR.